### PR TITLE
fix: upload success not properly cached

### PR DESCRIPTION
**Pull request by daytime-em**
Also:
* fixes `MuxUpload.Builder`'s setter methods not having an effect
* updates the Example App to have a pause/resume button again
* code-reformatting `MuxUpload`

### CODE REVIEW

File: app/src/main/java/com/mux/video/vod/demo/upload/Widgets.kt [hunk 1]
```diff
@@ -59,45 +72,49 @@ private fun AppBarInner(
...
-        }
 
-        if (startContent != null) {
-          Box(
-            modifier = Modifier
-              .align(Alignment.CenterStart)
-              .padding(16.dp)
-          ) {
-            startContent()
+          if (startContent != null) {
```
Comment (andrewjl-mux) [inline app/src/main/java/com/mux/video/vod/demo/upload/Widgets.kt:99]: Out of curiosity, what does `startContent` represent?

File: library/src/main/java/com/mux/video/upload/api/UploadStatus.kt [hunk 2]
```diff
@@ -35,19 +35,19 @@ sealed class UploadStatus {
...
-  object Started: UploadStatus()
+  data object Started: UploadStatus()
 
   /**
    * This upload is being prepared. If standardization is required, it is done during this step
    *
    * @see MuxUpload.Builder.standardizationRequested
    */
-  object Preparing: UploadStatus()
+  data object Preparing: UploadStatus()
```
Comment (andrewjl-mux) [inline library/src/main/java/com/mux/video/upload/api/UploadStatus.kt:50]: Is there any particular reason for why some `UploadStatus` are `data object` vs `class`? (e.g. `Uploading` for the latter.)

**Review by andrewjl-mux — APPROVED** [review]
LGTM, left some questions

**Comment by daytime-em** [comment]
Left some answers! Hope it's informative

**Pull request merged**